lightning/src/ln/onion_payment.rs

```diff
@@ -135,7 +161,9 @@ pub(super) fn create_fwd_pending_htlc_info(
 				reason: LocalHTLCFailureReason::InvalidOnionPayload,
 				err_data: Vec::new(),
 			}),
-		onion_utils::Hop::TrampolineForward { next_trampoline_hop_data, next_trampoline_hop_hmac, new_trampoline_packet_bytes, trampoline_shared_secret, .. } => {
+		onion_utils::Hop::TrampolineForward { ref outer_hop_data, next_trampoline_hop_data, next_trampoline_hop_hmac, new_trampoline_packet_bytes, trampoline_shared_secret, .. } => {
+			// TODO: return reason as forward issue, not as receiving issue when forwarding is ready.
```

Comment:
I'm not quite clear on what this `todo` means, still catching up on some context - @a-mpch could you help me out?